webrender/src/device/gfx.rs

```diff
@@ -2092,9 +2092,33 @@ impl<B: hal::Backend> Device<B> {
         let limits = adapter.physical_device.limits();
         let max_texture_size = 4400u32; // TODO use limits after it points to the correct texture size
 
-        let (device, queue_group) = adapter
-            .open_with::<_, hal::Graphics>(1, |family| surface.supports_queue_family(family))
-            .unwrap();
+        let (device, queue_group) = {
+            use hal::Capability;
+            use hal::queue::QueueFamily;
+
+            let family = adapter
+                .queue_families
+                .iter()
+                .filter(|family| {
+                    hal::Graphics::supported_by(family.queue_type())
+                        && surface.supports_queue_family(&family)
+                        && 1 <= family.max_queues()
+                })
+                .next().unwrap();
```

Comment:
why is it `filter(...).next().unwrap()` instead of just `find().unwrap()`?

Reply:
I have used the code of `open_with` from gfx-rs and I forgot to change the syntax here to match our case. Thanks for the  correction.